tensorflow/core/util/gpu_launch_config.h

```diff
@@ -27,6 +27,11 @@ limitations under the License.
 #include "tensorflow/core/platform/types.h"
 #include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
 
+#define FWD(func, alias) \
```

Comment:
Nice suggestion on the naming of the function.

Regarding the def should be within `#ifdef GOOGLE_CUDA`: My thoughts is that upstream ROCm build are still using the `Cuda*` related functions. If this forward declaration macro is CUDA only, the upstream build would fail.

Reply:
I believe that we have not upstreamed any ROCm code that references the `Cuda*` symbols from this file. If this assumption is incorrect, then you are right, the ROCm build will fail. But that is a good thing, since it will allow us to easily identify and fix all such instances (in the PR we will create to upstream this change). 

Our goal is to have the ROCm code refer to the `Gpu*` symbols only!

Reply:
Fair points! I will swap your suggestion of using `#ifdef GOOGLE_CUDA` with `#ifndef TENSORFLOW_USE_ROCM`, as a more conservative approach. I don't know anything beyond Cuda and ROCm build require, so will keep the alias on all builds outside ROCm.